Model trial training

Yamini-26 opened the pull request:
* Have made a few changes in the LECSFormer model code to bring it to working condition, installing all required packages and matching the versions of all packages. 
* Downloaded and used the 80 images with masks dataset from the NHA12D_A New Pavement Crack Dataset and a Comparison Study Of Crack Detection Algorithms paper.
* Divided the dataset into training and validation sets and created corresponding train.txt and val.txt base filenames.
* Some additional preprocessing steps to match with the dataset included, like - dimensionality reduction and image flipping, and array shape check to match the transpose operation in the np2Tensor.

ZhaoNan1 committed:
Initial commit

ZhaoNan1 committed:
LECSFormer

ZhaoNan1 committed:
images

ZhaoNan1 committed:
update

ZhaoNan1 committed:
Add files via upload

Yamini-26 committed:
LECSFormer Model Working

Yamini-26 committed:
Added the 80 images with masks for taining

Yamini-26 committed:
EasyBuild- Custom Python Env for CCR

chaozheUB commented on README.md:
```diff
@@ -1,2 +1,5 @@
+<<<<<<< HEAD
+<<<<<<< HEAD
```
you need to remove this.

chaozheUB commented on configs/config.yaml:
```diff
@@ -0,0 +1,8 @@
+    EMBED_DIM: 64
+    DEPTHS: [2, 2, 2, 2]
+    NUM_HEADS: [2, 4, 8, 16]
+    
```
make line 8 ane empty line.

chaozheUB commented on datasets/CrackLS80/images/APFD_01.jpg:
I suggest not adding the figures as part of the repo, but may be put it somewhere say on the box.
https://buffalo.box.com/s/7msenudm0yws4k17j24ym6dsznnd8v7n

chaozheUB reviewed (commented):
Please add a .gitignore and add `*.pyc` and `__pycache`.

Yamini-26 committed:
Removed __pycache__ from tracking

Yamini-26 committed:
Removed __pycache__ from tracking

Yamini-26 commented:
I've made all the suggested changes, could you please review it again?

chaozheUB commented on .gitignore:
```diff
@@ -0,0 +1 @@
+__pycache__/
```
please add one empty line to the end of the file.
Also you may need `*__pycache__/` as there are still files like this.

chaozheUB commented on datasets/CrackLS80/images/APFD_01.jpg:
As I mentioned previously, you want to move this dataset to the folder.
But please Include in the readme where to find them and how the training data folder structures be for the training code to run.

chaozheUB commented on requirements.txt:
```diff
@@ -0,0 +1,13 @@
+einops
+timm
+yacs
+# torch torchvision torchaudio --index-url https://download.pytorch.org/whl/cu118
```
add an emtpy line to the end.

chaozheUB replied:
^

chaozheUB commented on test.py:
```diff
@@ -0,0 +1,107 @@
+from collections import OrderedDict
+import cv2
+
+parser = argparse.ArgumentParser()
```
may be put this to a `__main__` function and put to the end?

chaozheUB replied:
Please add the parser to `__main__` function

chaozheUB commented on test.py:
```diff
@@ -0,0 +1,107 @@
+from networks.LECSFormer import LECSFormer
+from config import get_config
+from collections import OrderedDict
+import cv2
```
Please organize the import according to 
first group: default python package (sys, os, etc)
second group: installed package (numpy, pandas, torch)
third group: package in this repo

In each group, the package should be in alphabetical order.
You may ask chat gpt to reorganize this for you.

chaozheUB commented on train.py:
```diff
@@ -0,0 +1,80 @@
+from networks.LECSFormer import LECSFormer
+from trainer import trainer
+
+def parse_args():
```
same here.

chaozheUB commented on train.py:
```diff
@@ -0,0 +1,80 @@
+                        ).cuda()
+
+
+    trainer(args, model, config)
```
add an empty line to the end.

chaozheUB reviewed (commented):
Please also do a [flake8](https://flake8.pycqa.org/en/latest/) check on your program.
You need to install flake8 probably.
You may add a file name `.flake`  with the following to ignore
```
[flake8]
ignore = E261, E402, E501, W293, W504, F405, F403, E101
exclude = .git, .venv, build, cmake-build-*, **/temp/, *results/, requirements.txt
```

Yamini-26 committed:
refactor: code style, organize imports, relocate dataset

Yamini-26 committed:
README Update

Yamini-26 commented:
I've made all suggested changes - Included the flake8, resolved a huge bunch of existing flake8 errors, moved the dataset folder to the Box, updated the structure and usage instructions of the dataset in the README file, and sorted all the imports. 
Please let me know if it is all okay to proceed.

chaozheUB reviewed (commented):
Looks good! 
Are these codes able to run? 
If so I suggest you squash all the commits to one and we can merge this version

chaozheUB commented on try.py:
```diff
@@ -0,0 +1,107 @@
+# import os
+# import cv2
```
what does this script do?

Yamini-26 replied:
This is just to try things separately. I can remove this from the repo as well.

chaozheUB replied:
OK please remove unnecessary ones.

chaozheUB commented on output/crackls80/log.txt:
```diff
@@ -0,0 +1,32 @@
+[22:01:31.646] Namespace(root_path='datasets/CrackLS315', num_classes=1, output_dirs='output/crackls315', max_epochs=100, batch_size=16, n_gpu=2, deterministic=1, base_lr=0.001, img_size=[512, 512], seed=44, cfg='configs/config.yaml', resume=None, accumulation_steps=None, cache_mode='part', opts=None, zip=False, use_checkpoint=False, amp_opt_level='O1', tag=None, eval=False, throughput=False)
```
why do we need this?

Yamini-26 replied:
It's like a checkpoint to see which epoch exactly did the code stop running, or from where the model started getting overfit on the data, and so on.

chaozheUB replied:
OK, I assume there will be more like this in the future? Then you may want to have a log name with some date info,

Yamini-26 replied:
Yes, that's right. Since the training is not able to run on my system, due to insufficient GPU memory there is only one for now.

chaozheUB commented on networks/__pycache__/LECSFormer.cpython-311.pyc:
why this is still here?

Yamini-26 replied:
It's just the cached version, will remove that as well.

Yamini-26 committed:
refactor: code style, organize imports, relocate dataset

Yamini-26 commented:
Yes Professor, this version of code is able to run.

chaozheUB commented:
> Yes Professor, this version of code is able to run.

OK, then pleas squash to 2 commits and we can merge it.
If possible, you should have the first commit from the LECSFormer as it is, then the second commit contains your update. 
It may need some rework but for logging purpose it would worth it. After all we want the commit in master to carry clear meanings.

Yamini-26 commented:
> > Yes Professor, this version of code is able to run.
> 
> OK, then pleas squash to 2 commits and we can merge it. If possible, you should have the first commit from the LECSFormer as it is, then the second commit contains your update. It may need some rework but for logging purpose it would worth it. After all we want the commit in master to carry clear meanings.

Sure, I'll try and keep the LECSFormer as the first commit and add our changes as the second one.

Yamini-26 commented:
Hi professor, since some files and folder like try.py, and the dataset folder are not ignored in the previous version, but now that I have added all of those in the .gitignore, it is causing a conflict while trying to squash the commits. 
I keep getting this error - error: The following untracked working tree files would be overwritten by merge:
        try.py
Please move or remove them before you merge.
Aborting
hint: Could not execute the todo command
hint:
hint:     squash 570431836e8440a00f9f2674d9a4b9204f922b2f LECSFormer
hint:
hint: It has been rescheduled; To edit the command before continuing, please
hint: edit the todo list first:
hint:
hint:     git rebase --edit-todo
hint:     git rebase --continue

I tried 'git rm' to remove the cached versions, I pushed them all to 'stash', even though all those are clear, it still keeping throwing the same error.
So, I ran 'git clean -fd' to remove the files/directory that are untracked. This command removes all these files and folders in my working directory. Only then I am able to continue with the squashing. I would have to replace all these files and folders manually after squashing, and I also am not sure if this will occur next time as well while pushing or squashing. 
Can you please help me figure out what can be a permanent solution to this problem?

chaozheUB commented:
I am not familar with this error. You may have to look it up. Normally for such kind of activities, I would recommend the following.

1.  start an whole new branch from the current master, which should be empty.
2. manually added the raw code from the original repo that you start from, you should also add the .gitignore, .flake8 and so on. Make a first commit to this new branch
3. rebase your current branch `model-trial-training` on to this new branch, clean any conflicts.
4. squash all the commits that is on top of the new branch commit to 1 commit. 
 
This is only one way to do this.

chaozheUB commented on .flake8:
```diff
@@ -0,0 +1,3 @@
+[flake8]
+ignore = E261, E402, E501, W293, W504, F405, F403, E101, F841
+exclude = .git, .venv, build, cmake-build-*, **/temp/, *results/, requirements.txt
```
Please update this to your virtual environment name (`.venv` was the name for my case), and the corresponding folders you want to ignore the flake8.

chaozheUB commented on configs/config.yaml:
```diff
@@ -0,0 +1,7 @@
+  LECSFormer:
+    EMBED_DIM: 64
+    DEPTHS: [2, 2, 2, 2]
+    NUM_HEADS: [2, 4, 8, 16]  
```
Add a new line
